Draft-Accepted/Install-PowerShell.md

```diff
@@ -0,0 +1,217 @@
+---
+RFC: RFC00XX
+Author: Jason Helmick
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: Core
+Comments Due: 6/30/2022
+Plan to implement: Yes
+---
+
+# Install-PowerShell7 Cmdlet for Windows
+
+This RFC proposes the addition of the `Install-PowerShell7` cmdlet in Windows to perform a
```

Comment:
Some day there will be a PowerShell 8.  
`Install-PowerShell` COULD take a version and install downlevel versions, but should work one (e.g) 7.0 to get the 7.2.4 (today) or 7.3 when released. Like the currently suggested 
```
iex "& { $(irm https://aka.ms/install-powershell.ps1) } -UseMSI" 
``` 
but without the "get something random off the internet and run it" vibe  - in fact that code should be the starting point for this

Reply:
The Install-PowerShell.ps1 script here has worked well for several years!  It does everything (except install PWSH from the APP store) I would want. I am not clear on the motivation for a NEW installation script.

What does the Install-PowerShell.ps1 script _not_ do?

Rather than shipping a new command, why not just ship the existing one?

ANd +100000 for adding this script, whatever it is, to BOTH Windows PowerShell 5.1 and PowerShell 7. The latter would be easy to do while the former might require an updated Windows 11 (and Windows Server). Shipping via the PS Gallery would be a great way to deal with downlevel OSs as well.

Reply:
Agree with @doctordns  that the existing script has pretty much the desired functionality, and unless someone can create a significantly better wheel... I think that just as cumulative windows updates can deploy fixes _of_ Windows PowerShell 5.1 (in fact, IIRC the 5.0 to 5.1 release went out that way), an update _for_ 5.1 which is an extra script / module is possible it's just with 5.1 on support rather than active development there's been little to deliver.

Reply:
The current script gives current, preview and daily build, it doesn't allow a specific version to be requested, that's the only thing missing IMHO and half a dozen lines of code would add it.

Reply:
This is intended for only PowerShell 7